internal/commands/pull/command_test.go

```diff
@@ -317,6 +334,204 @@ Successfully pulled app down: app
+  "pancake": "blueberry",
+  "toast": "french"
+}`,
+				"template_2": `{
+  "egg": "scrambled",
+  "pancake": "plain",
+  "toast": "none"
+}`,
+			}
+			for templateID, expectedString := range expected {
+				destination := filepath.Join(profile.WorkingDirectory, "app", local.FrontendPath)
+
+				_, err = os.Stat(destination)
+				assert.Nil(t, err)
+
+				testData, readErr := ioutil.ReadFile(filepath.Join(destination, fmt.Sprintf("%s.json", templateID)))
+				assert.Nil(t, readErr)
+				assert.Equal(t, expectedString, string(testData))
+			}
+		})
+
+		t.Run("should return an error if resolving the template returns an error", func(t *testing.T) {
+			realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, bool, error) {
+				return nil, false, errors.New("some kind of error")
+			}
+
+			profile, teardown := mock.NewProfileFromTmpDir(t, "should_fail")
+			defer teardown()
+
+			out := new(bytes.Buffer)
+			ui := mock.NewUIWithOptions(mock.UIOptions{AutoConfirm: true}, out)
+
+			cmd := &Command{inputs{Project: "elsewhere", LocalPath: "app", TemplateID: "some-template-id"}}
+
+			err := cmd.Handler(profile, ui, cli.Clients{Realm: realmClient})
+			assert.Equal(t, "some kind of error", err.Error())
+		})
+		t.Run("should return an error if the template ID is not compatible with the app", func(t *testing.T) {
+			realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, bool, error) {
+				return []realm.Template{{ID: "wrong_id", Name: "should not appear"}}, true, nil
+			}
+
+			profile, teardown := mock.NewProfileFromTmpDir(t, "should_fail_also")
+			defer teardown()
+
+			out := new(bytes.Buffer)
+			ui := mock.NewUIWithOptions(mock.UIOptions{AutoConfirm: true}, out)
+
+			cmd := &Command{inputs{Project: "elsewhere", LocalPath: "app", TemplateID: "some-template-id"}}
+
+			err := cmd.Handler(profile, ui, cli.Clients{Realm: realmClient})
+			assert.Equal(t, "template some-template-id is not compatible with this app", err.Error())
+		})
+
+		t.Run("should return nothing and continue exporting the app if the app is not made with a template", func(t *testing.T) {
+			realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, bool, error) {
+				return nil, false, nil
+			}
+
+			profile, teardown := mock.NewProfileFromTmpDir(t, "should_fail_also")
+			defer teardown()
+
+			out := new(bytes.Buffer)
+			ui := mock.NewUIWithOptions(mock.UIOptions{AutoConfirm: true}, out)
+
+			cmd := &Command{inputs{Project: "elsewhere", LocalPath: "app"}}
+
+			err := cmd.Handler(profile, ui, cli.Clients{Realm: realmClient})
+			assert.Nil(t, err)
+
+			// Check to see that export went as normal
+			destination := filepath.Join(profile.WorkingDirectory, "app")
+
+			_, err = os.Stat(destination)
+			assert.Nil(t, err)
+
+			testData, readErr := ioutil.ReadFile(filepath.Join(destination, "test.json"))
+			assert.Nil(t, readErr)
+			assert.Equal(t, "{\"egg\":\"corn\"}\n", string(testData))
```

Comment:
[style nit] I want to say use backticks for string literals when writing valid json (or anything that needs `"`), but that forces you into adding a literal newline to this assert... so carry on as you see fit